### PR TITLE
refactor(storage): use static async trait in ColumnIterator

### LINKED ISSUE #394: storage: remove `async_trait` for `ColumnIterator` trait

### PR DESCRIPTION
**Pull request by zzl200012**
Fixed #394 

Signed-off-by: asuka <312856403@qq.com>

### CODE REVIEW

File: src/storage/secondary/column.rs [hunk 1]
```diff
@@ -57,22 +57,25 @@ pub trait ColumnBuilder<A: Array> {
...
+    type NextFuture<'a>: Future<Output = StorageResult<Option<(u32, A)>>> + 'a
+    where
+        Self: 'a;
+
     /// Get a batch and the starting row id from the column. A `None` return value means that
     /// there are no more elements from the block. By using `expected_size`, developers can
     /// get an array of NO MORE THAN the `expected_size` on supported column types.
-    async fn next_batch(
-        &mut self,
+    fn next_batch<'a>(
```
Comment (skyzh) [inline src/storage/secondary/column.rs:68]:
```suggestion
    fn next_batch(
```

No need to explicitly write `'a` here.

In the below parameters, simply use `&mut self`, `Option<&BitVec>`, and `Self::NextFuture<'_>`.

**Review by skyzh — COMMENTED** [review]
rest LGTM

File: src/storage/secondary/column.rs (hunk 1 above)
Comment (zzl200012) [reply src/storage/secondary/column.rs:68]: Sorry, I'm a beginner of rust. I tried this suggestion, but compiler said:
```
error[E0623]: lifetime mismatch
   --> src/storage/secondary/column/concrete_column_iterator.rs:298:10
    |
297 |         filter_bitmap: Option<&BitVec>,
    |                               ------- this parameter and the return type are declared with different lifetimes...
298 |     ) -> Self::NextFuture<'_> {
    |          ^^^^^^^^^^^^^^^^^^^^
    |          |
    |          ...but data from `filter_bitmap` is returned here
```
Is there any way to fix that?

Comment (skyzh) [reply src/storage/secondary/column.rs:68]: hmmm... Looks like a compiler bug. No need to fix.

**Review by skyzh — APPROVED** [review]

**Pull request merged**